### ISSUE TITLE
[Question] Error in the console and output

### ISSUE THREAD

**Issue by Tekbr** [redhat-developer/vscode-java#57]
--Sorry for the English, I used Google Translator --

If I have a project in Eclipse with the following project folders "`Folder - EAR"`, "`Folder - EJB`", "`Folder - WEB`"), I can not use this extension?

I say this because when I set up the `JAVA_HOME` and installed the extension. I only open my "`Folder - EJB`" with vscode the first time and open a `.java` file. The extension began to download the necessary files and I indicated that everything was "ok - :thumbsup: -  . For vscode the file was in error the file, but for the Eclipse was not. So I closed the vscode and open again and shows me  :-1: with the following message in the output:

> [Error - 18:48:31] 11/10/2016 18:48:21 Problem importing to workspace
> Invalid project description.
> org.eclipse.core.internal.resources.ResourceException: Invalid project description.
>   at org.eclipse.core.internal.resources.Project.checkDescription(Project.java:169)
>   at org.eclipse.core.internal.resources.Project.assertCreateRequirements(Project.java:53)
>   at org.eclipse.core.internal.resources.Project.create(Project.java:263)
>   at org.eclipse.core.internal.resources.Project.create(Project.java:247)
>   at org.jboss.tools.vscode.java.internal.managers.EclipseProjectImporter.createProject(EclipseProjectImporter.java:115)
>   at org.jboss.tools.vscode.java.internal.managers.EclipseProjectImporter.importToWorkspace(EclipseProjectImporter.java:85)
>   at org.jboss.tools.vscode.java.internal.managers.ProjectsManager.createProject(ProjectsManager.java:50)
>   at org.jboss.tools.vscode.java.internal.handlers.InitHandler$1.run(InitHandler.java:85)
>   at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
> Contains: OK
> Contains: c:\FOLDER\Folder 1\test\FolderEJB overlaps the location of another project: 'FolderEJB'

And the console (Toggle Developer Tools) shows this:

> shell.ts:428Illegal argument: Error: Illegal argument
>     at new e (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:9:24998)
>     at asHover (C:\Users\User \ .vscode\extensions\redhat.java-0.0.4\node_modules\vscode-languageclient\lib\protocolConverter.js:71:16)
>     at process._tickCallback (internal/process/next_tick.js:103:7)e.onUnexpectedError @ shell.ts:428
> 2shell.ts:428 Illegal argument: Error: Illegal argument
>     at new e (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:9:24998)
>     at asHover (C:\Users\User\ .vscode\extensions\redhat.java-0.0.4\node_modules\vscode-languageclient\lib\protocolConverter.js:71:16)
>     at process._tickCallback (internal/process/next_tick.js:103:7)e.onUnexpectedError @ shell.ts:428(anonymous function) @ shell.ts:341e.onUnexpectedError @ errors.ts:73u @ errors.ts:88(anonymous function) @ hover.ts:34m @ winjs.base.raw.js:1209B.enter @ winjs.base.raw.js:901n.Class.define._run @ winjs.base.raw.js:1068n.Class.define._error @ winjs.base.raw.js:1041m @ winjs.base.raw.js:1209B.enter @ winjs.base.raw.js:901n.Class.define._run @ winjs.base.raw.js:1068n.Class.define._error @ winjs.base.raw.js:1041(anonymous function) @ async.ts:358n.Class.define.done @ winjs.base.raw.js:1378m @ winjs.base.raw.js:1224B.enter @ winjs.base.raw.js:901n.Class.define._run @ winjs.base.raw.js:1068n.Class.define._error @ winjs.base.raw.js:1041e.resolveErr @ ipcRemoteCom.ts:110f @ ipcRemoteCom.ts:198_combinedTickCallback @ internal/process/next_tick.js:67_tickCallback @ internal/process/next_tick.js:98

and

> shell.ts:428 Illegal argument: Error: Illegal argument
>     at new e (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:9:24998)
>     at asHover (C:\Users\User \ .vscode\extensions\redhat.java-0.0.4\node_modules\vscode-languageclient\lib\protocolConverter.js:71:16)
>     at process._tickCallback (internal/process/next_tick.js:103:7)e.onUnexpectedError @ shell.ts:428(anonymous function) @ shell.ts:341e.onUnexpectedError @ errors.ts:73u @ errors.ts:88(anonymous function) @ hover.ts:34m @ winjs.base.raw.js:1209B.enter @ winjs.base.raw.js:901n.Class.define._run @ winjs.base.raw.js:1068n.Class.define._error @ winjs.base.raw.js:1041m @ winjs.base.raw.js:1209B.enter @ winjs.base.raw.js:901n.Class.define._run @ winjs.base.raw.js:1068n.Class.define._error @ winjs.base.raw.js:1041(anonymous function) @ async.ts:358n.Class.define.done @ winjs.base.raw.js:1378m @ winjs.base.raw.js:1224B.enter @ winjs.base.raw.js:901n.Class.define._run @ winjs.base.raw.js:1068n.Class.define._error @ winjs.base.raw.js:1041e.resolveErr @ ipcRemoteCom.ts:110f @ ipcRemoteCom.ts:198_combinedTickCallback @ internal/process/next_tick.js:67_tickCallback @ internal/process/next_tick.js:98

I saw something about maven project on the README, the as I am starting programming, I still have no idea to work the same. Sorry for anything.

**Comment by gorkem** [comment]
Do you have pom.xml files on your project? How is your project build?

**Comment by fbricon** [comment]
Looks like standalone Eclipse projects. 

@Tekbr can you share your project so we can try to reproduce the issue?

**Comment by Tekbr** [comment]
--Sorry for the English, I used Google Translator --

@gorkem, I do not have .pom.xml file, only .project and .classpath files.

If I understand correctly ...

@fbricon, sorry I can not now. But I created a test project you retracing the steps I think it is possible to reproduce the problem.

1- Using Java 8 and Eclipse Mars.
2 After opening the Eclipse add Wildfly Server 10.
3- Create EJB Project. Check the "Generate bj-jar.xml deploymet descriptor."
4- Create Dynamic Web Project. Check the "Generate web.xml deployment descriptor."
5- Create the EAR Application Project. Check in "Java EE module dependencies" EJB and Web folder. Also check "Generate application.xml deployment descriptor"
6- Create a package with a default class and save.
7-  Open throughout the project folder with VSCode or only EBJ folder. (Not yet open the .java file).
8- Install the extension and after installing open the .java file.
9- If the first time you start the project it shows the sign  :thumbsup: , but with an error in the file. Close vscode and open again (or use Reload Window). It makes recognition and shows the error already informed earlier with  :-1:.


Basically that's how it is set up my workspace.

And this test was done with the latest version of today 0.0.5 and 1.6.0 VSCode.

**Comment by lednhatkhanh** [comment]
I also have the same issue with a eclipse project: `org.eclipse.core.internal.resources.ResourceException: Invalid project description.` , so it's can;t import the eclipse projects?

**Comment by ywmail** [comment]
I have the some issue, I found that vscode  trying to import the Eclipse projects, as the `.project` and `.classpath` in the root folder, and it seems that eclipse JDT cannot import projects to the some folder.

so I move the eclipse project to a sub folder of the vs code work space, it works.

```
-- ECLIPSE ROOT
   |--- .classpath
   |--- .project
-- .vscode
-- others
```

**Comment by yaohaizh** [comment]
Close this issue as it is been quite a long time without furthur ado
